upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -218,6 +218,10 @@ func (e *AutoscalingGroup) Find(c *fi.Context) (*AutoscalingGroup, error) {
 			actual.MixedSpotAllocationStrategy = mpd.SpotAllocationStrategy
 			actual.MixedSpotInstancePools = mpd.SpotInstancePools
 			actual.MixedSpotMaxPrice = mpd.SpotMaxPrice
+			// MixedSpotMaxPrice must be set to "" in order to unset.
+			if mpd.SpotMaxPrice != nil && e.MixedSpotMaxPrice == nil {
+				e.MixedSpotMaxPrice = fi.String("")
```

Comment:
Why do this in `Find()`? Seems more like a thing to do in model.

Reply:
Then I think I'd have to convert the nil from AWS to "" here instead. This way seemed easier.

Reply:
It is easier, but makes things harder to track later. Find should only set expected for ignored fields.